rest_framework_json_api/utils.py

```diff
@@ -215,6 +215,8 @@ def get_related_resource_type(relation):
                 return get_related_resource_type(parent_model_relation)
 
     if relation_model is None:
+        if hasattr(relation, "child_relation"):
```

Comment:
not so sure this is a good spot for this fix as it is unexpected and many unnecessary checks have already been done on the relation to end up here.

What about adding a check on line 158 whether relation is instance of `ManyRelatedField` and if yes assign `relation = relation.child_relation`?

Reply:
I'm happy to consider other options.

Adding the check to 158 was my original solution but causes issues with resolving `ManyRelatedField`s on Polymorphic serializers. The test `example/tests/integration/test_polymorphism.py::test_polymorphism_on_included_relations` fails due to `relation.parent` resolving to the `ManyRelatedField` rather than the serializer that it is bound to, which results in the wrong `parent_model` being resolved.

Reply:
Hmm... the method `get_related_resource_type` is really messy and I think needs to have a better look how to better accomplish this. But I cannot see another easy way right now. So could you add at least a block comment above the if explaining why this is needed?

This will make it more comprehensible. Thanks.